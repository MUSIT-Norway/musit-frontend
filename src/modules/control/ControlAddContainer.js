
/*
 *  MUSIT is a museum database to archive natural and cultural history data.
 *  Copyright (C) 2016  MUSIT Norway, part of www.uio.no (University of Oslo)
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License,
 *  or any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License along
 *  with this program; if not, write to the Free Software Foundation, Inc.,
 *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import { connect } from 'react-redux';
import { addControl } from '../../reducers/control';
import { I18n } from 'react-i18nify';
import ControlAddComponent from './ControlAddComponent';
import { loadRoot } from '../../reducers/storageunit/grid';

const mapStateToProps = (state) => ({
  actor: state.auth.user.actor,
  translate: (key, markdown) => I18n.t(key, markdown),
  envReqData: state.storageGridUnit.root.data ? state.storageGridUnit.root.data.environmentRequirement : null,
  rootNode: state.storageGridUnit.root.data,
  user: state.auth.user
});

const mapDispatchToProps = (dispatch) => ({
  saveControl: (id, museumId, data, saveControlCallback) => {
    dispatch(addControl(id, data, {}, museumId, saveControlCallback));
  },
  loadStorageObj: (id, museumId) => {
    dispatch(loadRoot(id, museumId));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(ControlAddComponent);
