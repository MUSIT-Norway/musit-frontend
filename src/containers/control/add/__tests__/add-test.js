import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';
import React from 'react'
import ControlAddContainer from '../index'

function wrapWithContext(context, contextTypes, child){

  class WrapperWithContext extends React.Component {
    static childContextTypes = contextTypes;
    getChildContext() { return context; }
    render() {
      return child
    }
  }

  return <WrapperWithContext />;
}

describe('Render add control page', () => {
  it('should set default date and have correct date format', () => {
    const container = <ControlAddContainer
      doneDate={new Date(2016, 10, 10, 22).toISOString()}
      translate={(key) => key}
      params={{ }}
      saveControl={() => true}
      loadStorageObj={() => true}
      rootNode={{
        path: ',1,'
      }}
    />;
    const context = { closeModal: () => true };
    const contextTypes = { closeModal: React.PropTypes.func };
    const wrapper = shallow(
      wrapWithContext(context, contextTypes, container)
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
