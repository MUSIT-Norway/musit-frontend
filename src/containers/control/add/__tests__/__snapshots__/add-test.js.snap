exports[`Render add control page should set default date and have correct date format 1`] = `
<ControlAddContainer
  doneDate="2016-11-10T21:00:00.000Z"
  loadStorageObj={[Function loadStorageObj]}
  params={Object {}}
  rootNode={
    Object {
      "path": ",1,"
    }
  }
  saveControl={[Function saveControl]}
  translate={[Function translate]} />
`;
